packages/api/src/beacon/routes/beacon/block.ts

```diff
@@ -490,7 +499,16 @@ export function getDefinitions(config: ChainForkConfig): RouteDefinitions<Endpoi
           };
         },
         parseReqJson: ({body, headers, query}) => {
-          const fork = toForkName(fromHeaders(headers, MetaHeader.Version));
+          let fork: ForkName;
```

Comment:
On the publish v2 api the header is required, we don't need to apply the hack here

https://github.com/ethereum/beacon-APIs/blob/master/apis/beacon/blocks/blinded_blocks.v2.yaml#L43

The header is optional on v1 publishBlock / publishBlindedBlock because it was applied afterwards for ssz requests.

For all other apis, irrespective for JSON or SSZ, the headers must be present.